install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -54,9 +54,11 @@ spec:
         # in AppArmor.
         container.apparmor.security.beta.kubernetes.io/cilium-agent: "unconfined"
         container.apparmor.security.beta.kubernetes.io/clean-cilium-state: "unconfined"
+        {{- if .Values.cgroup.autoMount.enabled }}
         container.apparmor.security.beta.kubernetes.io/mount-cgroup: "unconfined"
         container.apparmor.security.beta.kubernetes.io/apply-sysctl-overwrites: "unconfined"
```

Comment:
I see why it makes sense for this config option to cover mount-cgroup, but I'm not sure I understand the relationship to `apply-sysctl-overwrites`. Isn't that for general newer systemd support?

Reply:
yup, apply-sysctl-overwrites is for newer systemd support, the issue is that we this init container is relying hostproc mount to run nsenter.